src/csvparser.cpp

```diff
@@ -4,3 +4,34 @@ CsvParser::CsvParser(QObject *parent) : QObject(parent)
 {
 
 }
+
+QStringList CsvParser::readAllLines(QString file_name_with_path)
+{
+    QFile file(file_name_with_path);
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+    {
+        return _allLines;
+    }
+
+    QTextStream in(&file);
+    while (!in.atEnd())
+    {
+        QString line = in.readLine();
+        _allLines.append(line);
+    }
+    return _allLines;
+}
+
+QList<QStringList> CsvParser::parseCsvData(QStringList csv_lines)
+{
+    QList <QStringList> parsedData;
+
+    QString cell;
+    foreach (cell, csv_lines)
+    {
+        QStringList row = cell.split(u',', Qt::SkipEmptyParts);
```

Comment:
Do you need to be concerned about text with a "," in quotes? For example a single cell might contain "a, b" which would be split here.

This could be something you plan to address later, just making sure its on your mind.

Reply:
There are multiple versions of CSV, such as one allowing the use of commas if the the entries themselves are enclosed within quotes; it really depends upon the way the CSV was generated. What I think I could do is, provide a dialog box on loading a CSV file which allows the user to specify the type of CSV intended to be imported. 

The options would be something like this:
1. Plain CSV.
2. Double Quotes Entries.
3. Single Quote Entries.
4. Custom Separator.

Then depending upon the option selected, the appropriate function could be called from the parser. I'll create an issue for this.